### PR TITLE
Add interaction-tracking/subscriptions

### PR DESCRIPTION
**Pull request by bvaughn**
Follow up to PR #13234

* Combine `enableInteractionTracking` and `enableInteractionTrackingObserver` feature flags
* Added `interaction-tracking/subscriptions` bundle
* Added multi-subscriber support
* Fixed existing bug with wrapped function not being passed params (and added test)

### CODE REVIEW

File: packages/interaction-tracking/src/InteractionTrackingSubscriptions.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import {enableInteractionTracking} from 'shared/ReactFeatureFlags';
+import {__subscriberRef} from 'interaction-tracking';
+
+let subscribers: Set<Subscriber> = (null: any);
+if (enableInteractionTracking) {
+  subscribers = new Set();
+}
+
+export function subscribe(subscriber: Subscriber): void {
+  subscribers.add(subscriber);
```
Comment (sebmarkbage) [inline packages/interaction-tracking/src/InteractionTrackingSubscriptions.js]: Shouldn't these be noops if `enableInteractionTracking` is false?

File: packages/interaction-tracking/src/InteractionTrackingSubscriptions.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      }
+    }
+  });
+
+  if (didCatchError) {
+    throw caughtError;
+  }
+}
+
+__subscriberRef.current = {
```
Comment (sebmarkbage) [inline packages/interaction-tracking/src/InteractionTrackingSubscriptions.js]: This shouldn't bother subscribing if `enableInteractionTracking` is false, right?

File: packages/interaction-tracking/src/InteractionTrackingSubscriptions.js (hunk 1 above)
Comment (bvaughn) [reply packages/interaction-tracking/src/InteractionTrackingSubscriptions.js]: Yikes. Yup

File: packages/interaction-tracking/src/InteractionTracking.js [hunk 3]
```diff
@@ -201,89 +184,77 @@ export function wrap(
...
-
-        return returnValue;
       } finally {
-        // Update pending async counts for all wrapped interactions.
-        // If this was the last scheduled async work for any of them,
-        // Mark them as completed.
-        wrappedInteractions.forEach(interaction => {
-          interaction.__count--;
+        try {
+          returnValue = callback.apply(undefined, arguments);
```
Comment (acdlite) [inline packages/interaction-tracking/src/InteractionTracking.js]: Just noticed this... `wrapped` is an arrow function here, so these are the arguments to the outer function. Need a test.

**Comment by react-linter** [comment]
<details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 5e0f073d509a74353533233706fce2e8044a0a15...71bddb67ebbbe89f5846c31c4f91c974940f9eeb</p>


  
## interaction-tracking
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
interaction-tracking.development.js | -12.2% | -6.8% | 8.2 KB | 7.2 KB | 2.33 KB | 2.17 KB | UMD_DEV
interaction-tracking.development.js | -12.5% | -7.0% | 8 KB | 7 KB | 2.26 KB | 2.1 KB | NODE_DEV
interaction-tracking-subscriptions.development.js | n/a | n/a | 0 B | 5.03 KB | 0 B | 1.43 KB | UMD_DEV
**interaction-tracking-subscriptions.production.min.js** | **n/a** | **n/a** | **0 B** | **1.55 KB** | **0 B** | **599 B** | **UMD_PROD**
interaction-tracking-subscriptions.development.js | n/a | n/a | 0 B | 4.77 KB | 0 B | 1.34 KB | NODE_DEV
**interaction-tracking-subscriptions.production.min.js** | **n/a** | **n/a** | **0 B** | **1.32 KB** | **0 B** | **502 B** | **NODE_PROD**

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/interaction-tracking/src/InteractionTracking.js (hunk 3 above)
Comment (bvaughn) [reply packages/interaction-tracking/src/InteractionTracking.js]: Good catch.

File: packages/interaction-tracking/src/InteractionTracking.js [hunk 4]
```diff
@@ -195,7 +195,7 @@ export function wrap(
     interaction.__count++;
   });
 
-  const wrapped = () => {
+  const wrapped = (...args) => {
```
Comment (acdlite) [inline packages/interaction-tracking/src/InteractionTracking.js]: You could also change this to a normal function :D

Comment (acdlite) [reply packages/interaction-tracking/src/InteractionTracking.js]: It's a nit. I'm just paranoid about the Babel output.

Comment (bvaughn) [reply packages/interaction-tracking/src/InteractionTracking.js]: The Babel output is fine. I've checked it. 😄

But sure, I don't mind.

**Pull request merged**